lib/repository.js

```diff
@@ -868,42 +868,67 @@ Repository.prototype.mergeBranches =
  * Goes through a rebase's rebase operations and commits them if there are
  * no merge conflicts
  *
- * @param {Repository}  repository  The repository that the rebase is being
- *                                  performed in
- * @param {Rebase}      rebase      The current rebase being performed
- * @param {Signature}   signature   Identity of the one performing the rebase
- * @return  {Int|Index} An error code for an unsuccesful rebase or an index for
- *                      a rebase with conflicts
+ * @param {Repository}  repository    The repository that the rebase is being
+ *                                    performed in
+ * @param {Rebase}      rebase        The current rebase being performed
+ * @param {Signature}   signature     Identity of the one performing the rebase
+ * @param {Function}    beforeNextFn  Callback to be called before each
+ *                                    invocation of next(). If the callback
+ *                                    returns a promise, the next() will be
+ *                                    called when the promise resolves.
+ * @return {Int|Index} An error code for an unsuccesful rebase or an index for
+ *                     a rebase with conflicts
  */
-function performRebase(repository, rebase, signature) {
-  return rebase.next()
-    .then(function(rebaseOperation, lel) {
-      return repository.openIndex()
-        .then(function(index) {
-          if (index.hasConflicts()) {
-            throw index;
-          }
+function performRebase(repository, rebase, signature, beforeNextFn) {
+  var beforeNextFnResult;
+
+  function getPromise() {
+    return rebase.next()
+      .then(function(rebaseOperation, lel) {
```

Comment:
This `lel` seems to have snuck its way in here during a previous PR. Could you remove it while you're here?